src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -1425,13 +1428,20 @@ private Type EmitRangeExprNode(AssociativeNode node)
                     op = new IntNode(-1);
                     break;
             }
+
+            bool hasStep = stepNode != null;
+            // The value of the dummy DoubleNode does not matter since the hasStep boolean will be false.
+            AssociativeNode dummyStepNode = AstFactory.BuildDoubleNode(1);
             var arguments = new List<AssociativeNode>
             {
                 fromNode,
                 toNode,
-                stepNode ?? new NullNode(),
+                // TODO_MSIL: Figure out a better solution for this scenario.
+                // Use DoubleNode(1) because standard replication cannot handle null to value type coerce
+                // The old VM handles builtin functions (like range expr) in a special way...that does not try coercion
```

Comment:
EmitFunctionCall could do something special for builtin nodes - as it does when it encounters operators or indexing calls...

Reply:
I will add a task for this case